Dockerfile.test

```diff
@@ -0,0 +1,18 @@
+FROM python:3.4
+
+RUN apt-get update \
+ && apt-get upgrade -y \
+ && apt-get install -y libgdal1-dev gdal-bin
```

Comment:
Recommendations based on Docker's [Best practices for writing Dockerfiles](http://docs-stage.docker.com/v1.10/engine/userguide/eng-image/dockerfile_best-practices/)
* [Sort multi-line arguments](http://docs-stage.docker.com/v1.10/engine/userguide/eng-image/dockerfile_best-practices/#sort-multi-line-arguments)
* [apt-get](http://docs-stage.docker.com/v1.10/engine/userguide/eng-image/dockerfile_best-practices/#apt-get)
  * You should avoid `RUN apt-get upgrade` or `dist-upgrade`, as many of the "essential" packages from the base images won’t upgrade inside an unprivileged container.
  * cleaning up the apt cache and removing `/var/lib/apt/lists` helps keep the image size down

```
RUN apt-get update \
  && apt-get upgrade -y \
  && apt-get install -y \ 
        gdal-bin \
        libgdal1-dev && \
  rm -rf /var/lib/apt/lists/*
```

Also, I'm not sure I would `apt-get upgrade` in the Dockerfile -- I'd let the upstream python image take care of that, so I might change this to:

```
RUN apt-get update \
  && apt-get install -y \
        gdal-bin \
        libgdal1-dev && \
    rm -rf /var/lib/apt/lists/*
```

and if we _really_ want to get picky, I find this style to be easier to read (possibly just personal preference?):

```
RUN apt-get update && \
    apt-get install -y \
      gdal-bin \
      libgdal1-dev && \
    rm -rf /var/lib/apt/lists/*
```